ironfish-cli/src/commands/wallet/multisig/sign.ts

```diff
@@ -0,0 +1,379 @@
+
+    if (flags.ledger) {
+      ledger = new Ledger(this.logger)
+      try {
+        await ledger.connect(true)
+      } catch (e) {
+        if (e instanceof Error) {
+          this.error(e.message)
+        } else {
+          throw e
+        }
+      }
+    }
+
+    let participantName: string
+    if (!flags.account) {
+      participantName = await ui.multisigSecretPrompt(client)
+    } else {
+      participantName = flags.account
+      const account = (await client.wallet.getAccounts()).content.accounts.find(
+        (a) => a === participantName,
+      )
+      if (!account) {
+        this.error(`Account ${participantName} not found`)
+      }
+    }
+
+    const unsignedTransactionInput =
+      flags.unsignedTransaction ??
+      (await ui.longPrompt('Enter the unsigned transaction', { required: true }))
+    const unsignedTransaction = new UnsignedTransaction(
+      Buffer.from(unsignedTransactionInput, 'hex'),
+    )
+    await renderUnsignedTransactionDetails(
+      client,
+      unsignedTransaction,
+      participantName,
+      this.logger,
+    )
+
+    const { commitment, identities } = await this.retryStep(async () => {
+      return this.performRound1(
+        client,
+        participantName,
+        unsignedTransaction,
+        unsignedTransactionInput,
+        ledger,
+      )
+    })
+
+    this.log('\n============================================')
+    this.log('\nCommitment:')
+    this.log(commitment)
+    this.log('\n============================================')
+
+    const willAggregateCommitments = await ui.confirmPrompt(
+      'Do you want to aggregate the commitments? (Needs to be performed by 1 participant)',
+    )
+
+    let signingPackage: string
+
+    if (willAggregateCommitments) {
+      signingPackage = await this.retryStep(() => {
+        return this.performAggregateCommitments(
+          client,
+          participantName,
+          commitment,
+          identities,
+          unsignedTransaction,
+          identities.length,
+        )
+      })
+
+      this.log('\n============================================')
+      this.log('\nSigning Package:')
+      this.log(signingPackage)
+      this.log('\n============================================')
+    } else {
+      signingPackage = await ui.longPrompt('Enter the signing package', { required: true })
+    }
+
+    const signatureShare = await this.retryStep(() =>
+      this.performCreateSignatureShare(
+        client,
+        participantName,
+        signingPackage,
+        unsignedTransaction,
+        ledger,
+      ),
+    )
+
+    this.log('\n============================================')
+    this.log('\nSignature Share:')
+    this.log(signatureShare)
+    this.log('\n============================================')
+
+    const willAggregateSignatures = await ui.confirmPrompt(
+      'Do you want to aggregate the signatures? (Needs to be performed by 1 participant)',
+    )
+
+    if (willAggregateSignatures) {
```

Comment:
It's also ok if everyone aggregates signature shares

Reply:
sounds good, removed